guides/source/7_1_release_notes.md

```diff
@@ -150,7 +150,51 @@ and reporting of the bulk enqueuing process.
 
 ### Composite primary keys
 
-TODO: Add description
+Applications can now create database tables with composite primary keys, and Rails will be able to derive these keys directly from the schema. This feature is particularly beneficial for many-to-many relationships and other complex data models where a single column is insufficient to uniquely identify a record.
+
+Active Record has been updated to fully support composite primary keys. All parts of the composite primary key will be included in the SQL generated by the `#reload`, `#update`, and `#delete methods`. Methods like `#first` and `#last` will use the full composite primary key in the `ORDER BY` statement.
+
+The `query_constraints`` macro can be used as a "virtual primary key" to achieve the same behaviors without modifying the database schema.
+
+Similarly, the `query_constraints:` option is accepted by associations to serve as a composite foreign key. This option configures the list of columns used for accessing the associated record.
+
+How to use:
```

Comment:
I would probably opt to take out this `How to use` section (let's move it to the guides?), and instead to provide a simple example for the `query_constraints` macro and a simple example of using `query_constraints` as an option on an association.

Reply:
If we *do* add a composite primary keys guide or large page, that would be an obvious place to point to.

Reply:
Done! Changed this big section in favor of a few small ones